nmpolicy/internal/expander/state_expander.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 NMPolicy Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package expander
+
+import (
+	"fmt"
+	"regexp"
+
+	yaml "sigs.k8s.io/yaml"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/types"
+)
+
+type StateExpander struct {
+	capResolver    CapturePathResolver
+	capturedStates map[string]types.CaptureState
+}
+
+type CapturePathResolver interface {
+	ResolveCapturePath(capturesCache map[string]types.CaptureState, capturePath string) (interface{}, error)
+}
+
+func New(capResolver CapturePathResolver) StateExpander {
+	return StateExpander{capResolver: capResolver}
+}
+
+func (c StateExpander) Expand(capturedStates map[string]types.CaptureState, marshaledDesiredState []byte) ([]byte, error) {
```

Comment:
```suggestion
func (c StateExpander) Expand(marshaledDesiredState []byte) ([]byte, error) {
```

The `capturedState` is already inside the `captureEntryPathResolver`